File: 拓展/C++/jifen.cpp
#include <iostream>
#include <functional>

double jifen(std::function<double(double)> f, double a, double b, int n) {
    double h = (b - a) / n;
    double total = 0.0;
    for (int i = 1; i < n; ++i) {
        total += f(a + i * h);
    }
    total = (0.5 * (f(a) + f(b)) + total) * h;
    return total;
}

double f(double x) {
    return 1.0;
}

int main() {
    std::cout << jifen(f, 1.0, 3.0, 1000) << std::endl;
    return 0;
}

File: 拓展/python/代码复习/jifen.py
def jifen(f,a,b,n):#a,b区间的f函数的积分，n越大越精细
    h=(b-a)/n
    total=sum(f(a+i*h) for i in range(1,n))
    total=(0.5*(f(a)+f(b))+total)*h
    return total

def f(x):
    return 1

print(jifen(f,1,3,1000))
